pkg/shares/blob_share_commitment_rules.go

```diff
@@ -45,35 +45,31 @@ func BlobSharesUsedNonInteractiveDefaults(cursor, squareSize, subtreeRootThresho
 
 // NextShareIndex determines the next index in a square that can be used. It
 // follows the blob share commitment rules defined in ADR-013. Assumes that all
-// args are non negative, and that squareSize is a power of two.
+// args are non negative, that squareSize is a power of two and that the blob can
+// fit in the square. The cursor is expected to be the index after the end of
+// the previous blob.
 //
-// https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules
+// See https://github.com/celestiaorg/celestia-app/blob/main/specs/src/specs/data_square_layout.md
+// for more information.
 func NextShareIndex(cursor, blobShareLen, squareSize, subtreeRootThreshold int) int {
 	// if we're starting at the beginning of the row, then return as there are
 	// no cases where we don't start at 0.
 	if isStartOfRow(cursor, squareSize) {
 		return cursor
 	}
```

Comment:
[optional] this optimization doesn't seem necessary. All tests pass with it's removal and the specs don't describe it.
```suggestion
```

Reply:
Yeah it's a minor optimisation. I don't mind removing it